django/db/models/sql/query.py

```diff
@@ -1112,16 +1112,16 @@ def check_related_objects(self, field, value, opts):
                 for v in value:
                     self.check_query_object_type(v, opts, field)
 
-    def check_filterable(self, expression):
+    def _check_filterable(self, expression):
         """Raise an error if expression cannot be used in a WHERE clause."""
-        if not getattr(expression, 'filterable', 'True'):
+        if not getattr(expression, 'filterable', True):
             raise NotSupportedError(
                 expression.__class__.__name__ + ' is disallowed in the filter '
                 'clause.'
             )
         if hasattr(expression, 'get_source_expressions'):
             for expr in expression.get_source_expressions():
-                self.check_filterable(expr)
+                self._check_filterable(expr)
```

Comment:
Is there any reason reason to rename `_check_filterable` to `check_filterable`?

Reply:
> Is there any reason reason to rename _check_filterable to check_filterable?

The other way around you mean?
I think the fact it's public was my oversight, it is really an internal method. 
But it does not affect the bugfix itself, so can be omitted or moved to a separate commit.

Reply:
IMO it's more an internal method but I don't see much harm in keeping it _public_.

Reply:
Ok, left as is.